docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -88,142 +158,155 @@ Create the `package.json` file in the root of your project and add the following
+This function looks like a regular TypeScript function. With the Temporal TypeScript SDK, you define Activities using an exportable TypeScript module.
 
-:::
+The response from `icanhazip.com` is plain-text, and it includes a newline, so you trim off the newline character before returning the result.
 
-Next, add the following TypeScript code to define the Workflow:
+Notice that there's no error-handling code in this function. When you build your Workflow, you''ll use Temporal's Activity Retry policies to retry this code automatically if there's an error.
 
-<!--SNIPSTART typescript-hello-workflow-->
-[hello-world/src/workflows.ts](https://github.com/temporalio/samples-typescript/blob/main/hello-world/src/workflows.ts)
-```ts
-import { proxyActivities } from '@temporalio/workflow';
-// Only import the activity types
-import type * as activities from './activities';
-
-const { greet } = proxyActivities<typeof activities>({
-  startToCloseTimeout: '1 minute',
-});
+Now add the second Activity that accepts an IP address and retrives location data. In In `src/activities.ts`, add the following code to define it:
 
-/** A workflow that simply calls an activity */
-export async function example(name: string): Promise<string> {
-  return await greet(name);
+```ts
+// Use the IP address to get the location.
+export async function getLocationInfo(ip: string): Promise<string> {
+  const url = `http://ip-api.com/json/${ip}`;
+  const response = await fetch(url);
+  const data = await response.json();
+  return `${data.city}, ${data.regionName}, ${data.country}`;
 }
 ```
-<!--SNIPEND-->
 
-In this code, the variable `greet` is assigned the value of `proxyActivites`, which is a method from the Temporal TypeScript SDK that lets you configure the Activity with different options. In this example, you have specified that the Start-to-Close Timeout for your Activity will be one minute, meaning that your Activity has one minute to begin before it times out. Of all the Temporal timeout options, `startToCloseTimeOut` is the one you should always set.
+This Activity follows the same pattern as the `getIP` Activity. It's an exported async function that uses `fetch` to call a remote service. This time, the service returns JSON data rather than text.
 
-The `example` function executes an Activity called `greet`, and the function returns the result of the Activity. You'll create the `greet` function in the next section.
+While Activities can accept input argument, it's a best practice to send a single argument rather than multiple arguments. In this case you only have a single String. If you had more than one argument, you should bundle them up in a serializable object.  Review the [Activity parameters](https://docs.temporal.io/dev-guide/typescript/foundations/#activity-parameters) section of the Temporal documentation for more details, as there are some limitations you'll want to be aware of when running more complex applications.
```

Comment:
```suggestion
While Activities can accept input arguments, it's a best practice to send a single argument rather than multiple arguments. In this case you only have a single String. If you had more than one argument, you should bundle them up in a serializable object.  Review the [Activity parameters](https://docs.temporal.io/dev-guide/typescript/foundations/#activity-parameters) section of the Temporal documentation for more details, as there are some limitations you'll want to be aware of when running more complex applications.
```